test/db.tests.js

```diff
@@ -986,4 +989,102 @@ describe('LimitDBRedis', () => {
       });
     });
   });
-});
+
+  describe('PING', () => {
+    let ping = {
+      enabled: true,
+      interval: 10,
+      maxFailedAttempts: 3,
+      reconnectIfFailed: true
+    }
+  
+    let redisProxy;
+    let toxiproxy;
+  
+    beforeEach((done) => {
+      toxiproxy = new Toxiproxy("http://localhost:8474");
+      proxyBody = {
+        listen: "0.0.0.0:22222",
+        name: "ihsw_test_redis_master",
+        upstream: "redis:6379"
+      };
+      toxiproxy.createProxy(proxyBody)
+      .then((proxy) => {
+        redisProxy = proxy; 
+        done();
+      })
+      
+    })
+  
+    afterEach((done) => {
+      redisProxy.remove().then(done)
+    })
+  
+    it('should emit ping success', (done) => {
+      let called = false;
+      db = new LimitDB({ uri: 'localhost:22222', buckets, prefix: 'tests:', ping })
+      db.on(('ping - success'), () => {
+        if (!called) {
+          called = true;
+          done();
+        }
+      });
+    });
+  
+    it('should emit "ping - fail" when redis stops responding pings and client is not configured to reconnect', (done) => {
+      let called = false;
+      db = new LimitDB({ uri: 'localhost:22222', buckets, prefix: 'tests:', ping: {...ping, reconnectIfFailed: false} })
+      db.on(('ready'), () => {
+        redisProxy.enabled = false
+        redisProxy.update()
+      })
+      db.on(('ping - fail'), () => {
+        if (!called) {
+          called = true;
+          done();
+        }
+      });
+    });
+  
+    it('should emit "ping - reconnect attempted" when redis stops responding pings and client is configured to reconnect', (done) => {
+      let called = false;
+      db = new LimitDB({ uri: 'localhost:22222', buckets, prefix: 'tests:', ping })
+      db.on(('ready'), () => {
+        redisProxy.enabled = false
+        redisProxy.update()
+      })
+      db.on(('ping - reconnect attempted'), () => {
+        if (!called) {
+          called = true;
+          done();
+        }
+      });
+    });
+  
+    it('should NOT emit ping events when ping is disabled', (done) => {
+      let called = false;
+      db = new LimitDB({ uri: 'localhost:22222', buckets, prefix: 'tests:', ping: {...ping, enabled: false} })      
+      db.on(('ping - success'), () => {
+        if (!called) {
+          called = true;
+          done(new Error('event `ping - success` emitted'));
+        }
+      });
+      db.on(('ping - fail'), () => {
+        if (!called) {
+          called = true;
+          done(new Error('event `ping - fail` emitted'));
+        }
+      });
+      db.on(('ping - reconnect attempted'), () => {
+        if (!called) {
+          called = true;
+          done(new Error('event `ping - reconnect attempted` emitted'));
+        }
+      });
+      //If after 100ms there are no interactions, we mark the test as passed.
+      setTimeout(done, 100)
+    });
+  });
```

Comment:
Can we add a test verifying the redis reconnection. I mean something as:
1. Establish the connection
2. Call any redis operation and verify it works
3. Enable toxiProxy
4. Reach `ping - reconnect attempted``
5. Call any redis operation and verify it doesn't work
6. Disable the toxiProxy
7. Verify reception of  `ping - success`
8. Call any redis operation and verify it works

It doesn't need to be all these steps. But I'm worried about `this.redis.disconnect(true);` . I'd like to add a test to verify it will reconnect successfully